### PR TITLE
BaSP-TG-19: Task seeds and tests

### PR DESCRIPTION
**Pull request by anapaulib**

### CODE REVIEW

File: src/test/tasks.test.js [hunk 1]
```diff
@@ -1,3 +1,112 @@
...
+    const res = await request(app).get(`/tasks/${correctMockedTaskId}`).send();
+    expect(res.status).toBe(200);
+    expect(res.body.error).toBeFalsy();
+    expect(res.body.data).toBeDefined();
+    expect(res.body.message).toBe('Tasks found');
+    expect(res.body.data).toMatchObject({
+      _id: correctMockedTaskId,
+    });
+  });
+  test('Should return status code 404 when you send a non existing id', async () => {
```
Comment (disimoneleandro) [inline src/test/tasks.test.js:39]: it is recommended to leave a space between test and test

File: src/test/tasks.test.js [hunk 2]
```diff
@@ -1,3 +1,112 @@
...
+    const res = await request(app).get(`/tasks/${wrongMockedTaskId}`).send();
+    expect(res.status).toBe(404);
+    expect(res.body.error).toBeFalsy();
+    expect(res.body.message).toBe('Task does not exists');
+  });
+  test('Should return status code 400 when you send an invalid id', async () => {
```
Comment (disimoneleandro) [inline src/test/tasks.test.js:45]: the same as above

File: src/test/tasks.test.js [hunk 3]
```diff
@@ -1,3 +1,112 @@
...
+});
+
+describe('DELETE /tasks/:id', () => {
+  test('Should delete a task', async () => {
+    const res = await request(app).delete(`/tasks/${correctMockedTaskId}`).send();
+    expect(res.status).toBe(200);
+    expect(res.body.error).toBeFalsy();
+    expect(res.body.message).toBe(`Task with Id ${correctMockedTaskId} deleted`);
+  });
+  test('Should return status code 404 when you send a non existing id', async () => {
```
Comment (disimoneleandro) [inline src/test/tasks.test.js:59]: the same as above

File: src/test/tasks.test.js [hunk 4]
```diff
@@ -1,3 +1,112 @@
...
+    const res = await request(app).delete(`/tasks/${wrongMockedTaskId}`).send();
+    expect(res.status).toBe(404);
+    expect(res.body.error).toBeFalsy();
+    expect(res.body.message).toBe(`Task with Id ${wrongMockedTaskId} does not exists`);
+  });
+  test('Should return status code 400 when you send an invalid id', async () => {
```
Comment (disimoneleandro) [inline src/test/tasks.test.js:65]: the same as above

File: src/test/tasks.test.js [hunk 5]
```diff
@@ -1,3 +1,112 @@
...
+});
+
+describe('POST /tasks', () => {
+  test('Should create a task', async () => {
+    const res = await request(app).post('/tasks').send(mockedTask);
+    expect(res.status).toBe(201);
+    expect(res.body.error).toBeFalsy();
+    expect(res.body.message).toBe('Task created successfully');
+  });
+  test('Should return status code 400 when the task is not created', async () => {
```
Comment (disimoneleandro) [inline src/test/tasks.test.js:79]: the same as above

File: src/test/tasks.test.js [hunk 6]
```diff
@@ -1,3 +1,112 @@
...
+    expect(res.body.data).toBeDefined();
+    // eslint-disable-next-line no-underscore-dangle
+    expect(res.body.message).toBe(`Task with Id ${tasks[0]._id} updated`);
+    expect(res.body.data).toMatchObject({
+      // eslint-disable-next-line no-underscore-dangle
+      _id: tasks[0]._id,
+      description: mockedTask.description,
+    });
+  });
+  test('Should return status code 404 when you send a non existing id', async () => {
```
Comment (disimoneleandro) [inline src/test/tasks.test.js:101]: the same as above

File: src/test/tasks.test.js [hunk 7]
```diff
@@ -1,3 +1,112 @@
-test('', () => {
-  expect(2).toBe(2);
+import request from 'supertest';
+import app from '../app';
+import Tasks from '../models/Tasks';
+import tasks from '../seed/tasks';
+
```
Comment (DamianPalavecino) [inline src/test/tasks.test.js:5]:
```suggestion
describe('Endpoints Test for /tasks', () => {
```
Here you can add one more describe method which will contain all of the code below

File: src/test/tasks.test.js [hunk 8]
```diff
@@ -1,3 +1,112 @@
...
+const correctMockedTaskId = '63568fcc930a506d2400e17f';
+const wrongMockedTaskId = '5355c950520cf5e99219bb5a';
+const wrongTaskId = 5;
+
+const mockedTask = {
+  description: 'Mocked task description',
+};
+
+describe('GET /tasks', () => {
+  test('Should return the list of tasks', async () => {
```
Comment (GinaSchP) [inline src/test/tasks.test.js:19]:
```suggestion
  test('Should return status code 200 and the list of tasks', async () => {
```

File: src/test/tasks.test.js [hunk 9]
```diff
@@ -1,3 +1,112 @@
...
+    const res = await request(app).get('/tasks').send();
+    expect(res.status).toBe(200);
+    expect(res.body.error).toBeFalsy();
+    expect(res.body.data).toBeDefined();
+    expect(res.body.message).toBe('Tasks found');
+  });
+});
+
+describe('GET /tasks/:id', () => {
+  test('Should get a task', async () => {
```
Comment (GinaSchP) [inline src/test/tasks.test.js:29]: same as above

File: src/test/tasks.test.js [hunk 10]
```diff
@@ -1,3 +1,121 @@
...
+
+  describe('POST /tasks', () => {
+    test('Should return a status code 201 and create a task', async () => {
+      const res = await request(app).post('/tasks').send(mockedTask);
+      expect(res.status).toBe(201);
+      expect(res.body.error).toBeFalsy();
+      expect(res.body.message).toBe('Task created successfully');
+    });
+
+    test('Should return status code 400 when the task is not created', async () => {
```
Comment (lmalvasia) [inline src/test/tasks.test.js:85]: You should add a correct description with the purpose of why this is going to fail.
```suggestion
    test('should return status code 400 when the request body is empty', async () => {
```

**Review by lmalvasia — APPROVED** [review]

**Pull request merged**